integrationTests/relayers/slowTests/framework/testSetup.go

```diff
@@ -242,24 +238,44 @@ func (setup *TestSetup) isTransferDoneFromEthereumForToken(sender KeysHolder, re
 
 func (setup *TestSetup) checkHolderMvxBalanceForToken(holder KeysHolder, isSender bool, params TestTokenParams) bool {
 	setup.mutBalances.Lock()
+	defer setup.mutBalances.Unlock()
 
 	addr := holder.MvxAddress.String()
 	balanceMapping, exists := setup.mvxBalances[addr]
 	if !exists {
 		return false
 	}
-	expectedBalance := big.NewInt(0).Set(balanceMapping[params.AbstractTokenIdentifier])
-	holderName := addressToName[addr]
-	transferAmounts := params.EthTestAddrsExtraBalances[holderName]
 
-	setup.mutBalances.Unlock()
+	actualBalance := setup.MultiversxHandler.GetESDTUniversalTokenBalance(setup.Ctx, holder.MvxAddress, params.AbstractTokenIdentifier)
 
-	expectedBalance.Add(expectedBalance, transferAmounts[1])
-	if isSender {
-		expectedBalance.Add(expectedBalance, transferAmounts[0])
+	return setup.checkHolderBalanceForTokenHelper(balanceMapping, params, actualBalance, addr, isSender)
+}
+
+func (setup *TestSetup) checkHolderEthBalanceForToken(holder KeysHolder, isSender bool, params TestTokenParams) bool {
+	setup.mutBalances.Lock()
+	defer setup.mutBalances.Unlock()
+
+	addr := holder.EthAddress.String()
+	balanceMapping, exists := setup.ethBalances[addr]
+	if !exists {
+		return false
 	}
 
-	actualBalance := setup.MultiversxHandler.GetESDTUniversalTokenBalance(setup.Ctx, holder.MvxAddress, params.AbstractTokenIdentifier)
+	actualBalance := setup.EthereumHandler.GetBalance(holder.EthAddress, params.AbstractTokenIdentifier)
+
+	return setup.checkHolderBalanceForTokenHelper(balanceMapping, params, actualBalance, addr, isSender)
+}
+
+func (setup *TestSetup) checkHolderBalanceForTokenHelper(balanceMapping map[string]*big.Int, params TestTokenParams, actualBalance *big.Int, addr string, isSender bool) bool {
+	holderName := setup.AddressToName[addr]
```

Comment:
if the mutex suggestion is applied, this function will need its mutex protection